src/ngx_http_lua_log.c

```diff
@@ -313,4 +314,107 @@ ngx_http_lua_inject_log_consts(lua_State *L)
     /* }}} */
 }
 
+
+#ifndef NGX_LUA_NO_FFI_API
+int
+ngx_http_lua_ffi_filter_log(int level, u_char *err, size_t *errlen)
+{
+#ifdef HAVE_INTERCEPT_ERROR_LOG_PATCH
+    ngx_http_lua_log_ringbuff_t     *log_ringbuff;
+
+    log_ringbuff = ngx_cycle->intercept_error_log_data;
+
+    if (!log_ringbuff) {
+        *errlen = ngx_snprintf(err, *errlen,
+                               "API \"ngx.filter_log\" depends on directive "
+                               "\"lua_intercept_error_log\"")
+                  - err;
+        return NGX_ERROR;
+    }
+
+    if (level > NGX_LOG_DEBUG || level < NGX_LOG_STDERR) {
+        *errlen = ngx_snprintf(err, *errlen, "bad log level: %d", level)
+                  - err;
+        return NGX_ERROR;
+    }
+
+    log_ringbuff->filter_level = level;
+    return NGX_OK;
+#else
+    *errlen = ngx_snprintf(err, *errlen,
+                           "missing intercept error log patch")
+              - err;
+    return NGX_ERROR;
+#endif
+}
+
+
+int
+ngx_http_lua_ffi_errlog_count(u_char *err, size_t *errlen)
+{
+#ifdef HAVE_INTERCEPT_ERROR_LOG_PATCH
+    ngx_http_lua_log_ringbuff_t     *log_ringbuff;
```

Comment:
We should use `ringbuf` instead of `ringbuff` in identifiers. Please fix other similar places.